x-pack/plugins/security_solution/public/cases/components/status/button.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo, useCallback, useState, useMemo, useEffect } from 'react';
+import { EuiButton } from '@elastic/eui';
+
+import { CaseStatuses, caseStatuses } from '../../../../../case/common/api';
+import { statuses } from './config';
+
+interface Props {
+  status: CaseStatuses;
+  disabled: boolean;
+  isLoading: boolean;
+  onStatusChanged: (status: CaseStatuses) => void;
+}
+
+const getNextItem = (item: number) => (item + 1) % caseStatuses.length;
+
+const StatusActionButtonComponent: React.FC<Props> = ({
+  status,
+  onStatusChanged,
+  disabled,
+  isLoading,
+}) => {
+  // Rotate over the array of statuses. open -> in-progress -> closes -> open...
+  const indexOfCurrentStatus = useMemo(
+    () => caseStatuses.findIndex((caseStatus) => caseStatus === status),
+    [status]
+  );
+  const [nextStatusIndex, setNextStatusIndex] = useState(getNextItem(indexOfCurrentStatus));
+
+  // The useEffect is needed to update status updates from the context menu.
+  useEffect(() => setNextStatusIndex(getNextItem(indexOfCurrentStatus)), [indexOfCurrentStatus]);
```

Comment:
```suggestion
  const nextStatusIndex = useMemo(() => getNextItem(indexOfCurrentStatus), [indexOfCurrentStatus]);
```
This should work, right?

Reply:
or not because of your onClick?

Reply:
or does `onStatusChanged` change the `indexOfCurrentStatus` which would update the `useMemo`??

Reply:
The `onClick` needs the state. `onStatusChanged` does not change the `indexOfCurrentStatus`.

Reply:
I was wrong. It can be done as you said!